drivers/npmx/npmx_shell.c

```diff
@@ -311,34 +311,402 @@ static npmx_instance_t *npmx_instance_get(const struct shell *shell)
 	return npmx_instance;
 }
 
-static bool check_instance_index(const struct shell *shell, const char *instance_name,
-				 uint32_t index, uint32_t max_index)
+static bool check_instance_index(const struct shell *shell, const char *instance_name,
+				 uint32_t index, uint32_t max_index)
+{
+	if (index >= max_index) {
+		shell_error(shell, "Error: %s instance index is too high: no such instance.",
+			    instance_name);
+		return false;
+	}
+
+	return true;
+}
+
+static npmx_adc_t *adc_instance_get(const struct shell *shell)
+{
+	npmx_instance_t *npmx_instance = npmx_instance_get(shell);
+
+	return npmx_instance ? npmx_adc_get(npmx_instance, 0) : NULL;
+}
+
+static npmx_charger_t *charger_instance_get(const struct shell *shell)
+{
+	npmx_instance_t *npmx_instance = npmx_instance_get(shell);
+	if (npmx_instance == NULL) {
+		return NULL;
+	}
+
+	return npmx_instance ? npmx_charger_get(npmx_instance, 0) : NULL;
```

Comment:
I would compare `npmx_instance ` to `NULL`, but it is a minor thing, can be omitted.

Reply:
It will increase the lines of the code after the code format. For example gpio:
```
	return (npmx_instance && status) ? npmx_gpio_get(npmx_instance, (uint8_t)index) : NULL;
```
```
	return ((npmx_instance != NULL) && status) ? npmx_gpio_get(npmx_instance, (uint8_t)index) :
						     NULL;
```